pulsar-client-api/src/main/java/org/apache/pulsar/client/api/transaction/Transaction.java

```diff
@@ -29,6 +29,16 @@
 @InterfaceStability.Evolving
 public interface Transaction {
 
+    enum State {
+        OPEN,
+        COMMITTING,
+        ABORTING,
+        COMMITTED,
+        ABORTED,
+        ERROR,
```

Comment:
It's better to add a document for `ERROR`
Or can we rename it to `FAILED`?

Reply:
`FAILED` can be retried. The three reasons for this state are not retrievable, so `ERROR` may be more accurate